test/cmd/legacy-script.sh

```diff
@@ -337,43 +337,43 @@ runTests() {
-  port_field="(index .spec.ports 0).port"
-  port_name="(index .spec.ports 0).name"
-  second_port_field="(index .spec.ports 1).port"
-  second_port_name="(index .spec.ports 1).name"
-  image_field="(index .spec.containers 0).image"
-  pod_container_name_field="(index .spec.containers 0).name"
-  container_name_field="(index .spec.template.spec.containers 0).name"
-  hpa_min_field=".spec.minReplicas"
-  hpa_max_field=".spec.maxReplicas"
-  hpa_cpu_field=".spec.targetCPUUtilizationPercentage"
-  template_labels=".spec.template.metadata.labels.name"
-  statefulset_replicas_field=".spec.replicas"
-  statefulset_observed_generation=".status.observedGeneration"
-  job_parallelism_field=".spec.parallelism"
-  deployment_replicas=".spec.replicas"
-  secret_data=".data"
-  secret_type=".type"
-  change_cause_annotation='.*kubernetes.io/change-cause.*'
-  pdb_min_available=".spec.minAvailable"
-  pdb_max_unavailable=".spec.maxUnavailable"
-  generation_field=".metadata.generation"
-  template_generation_field=".spec.templateGeneration"
-  container_len="(len .spec.template.spec.containers)"
-  image_field0="(index .spec.template.spec.containers 0).image"
-  image_field1="(index .spec.template.spec.containers 1).image"
+  export labels_field=".metadata.labels"
+  export annotations_field=".metadata.annotations"
+  export service_selector_field=".spec.selector"
+  export rc_replicas_field=".spec.replicas"
+  export rc_status_replicas_field=".status.replicas"
+  export rc_container_image_field=".spec.template.spec.containers"
+  export rs_replicas_field=".spec.replicas"
+  export port_field="(index .spec.ports 0).port"
+  export port_name="(index .spec.ports 0).name"
+  export second_port_field="(index .spec.ports 1).port"
+  export second_port_name="(index .spec.ports 1).name"
+  export image_field="(index .spec.containers 0).image"
+  export pod_container_name_field="(index .spec.containers 0).name"
+  export container_name_field="(index .spec.template.spec.containers 0).name"
+  export hpa_min_field=".spec.minReplicas"
+  export hpa_max_field=".spec.maxReplicas"
+  export hpa_cpu_field=".spec.targetCPUUtilizationPercentage"
+  export template_labels=".spec.template.metadata.labels.name"
+  export statefulset_replicas_field=".spec.replicas"
+  export statefulset_observed_generation=".status.observedGeneration"
+  export job_parallelism_field=".spec.parallelism"
+  export deployment_replicas=".spec.replicas"
+  export secret_data=".data"
+  export secret_type=".type"
+  export change_cause_annotation='.*kubernetes.io/change-cause.*'
+  export pdb_min_available=".spec.minAvailable"
+  export pdb_max_unavailable=".spec.maxUnavailable"
+  export generation_field=".metadata.generation"
+  export template_generation_field=".spec.templateGeneration"
+  export container_len="(len .spec.template.spec.containers)"
+  export image_field0="(index .spec.template.spec.containers 0).image"
+  export image_field1="(index .spec.template.spec.containers 1).image"
```

Comment:
why do we have so many exports here?

Reply:
Check failed with "SC2034: image_field1 appears unused. Verify use (or export if used externally)."
All of them are used externally and I think export them should be the case we need.

Reply:
where are they used?

Reply:
They are used in some other files under `test/cmd/`.
``` base
# grep "image_field1" -rn test/cmd/*
test/cmd/apps.sh:78:  kube::test::wait_object_assert daemonset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DAEMONSET_R2_2}:"
test/cmd/apps.sh:84:  kube::test::get_object_assert daemonset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DAEMONSET_R2_2}:"
test/cmd/apps.sh:98:  kube::test::wait_object_assert daemonset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DAEMONSET_R2_2}:"
test/cmd/apps.sh:339:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:343:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:349:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:353:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:357:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:361:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
test/cmd/apps.sh:365:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
test/cmd/apps.sh:430:  kube::test::wait_object_assert statefulset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PAUSE_V2}:"
test/cmd/apps.sh:436:  kube::test::get_object_assert statefulset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PAUSE_V2}:"
test/cmd/apps.sh:450:  kube::test::wait_object_assert statefulset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PAUSE_V2}:"
test/cmd/core.sh:1267:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
```
If it make more sense that I will add one comments above these declares to explain why we need to export them.

Reply:
thanks!

sorry I'm so behind reviewing this